pkg/gui/presentation/commits.go

```diff
@@ -312,9 +312,32 @@ func displayCommit(
 	bisectInfo *git_commands.BisectInfo,
 	isYouAreHereCommit bool,
 ) []string {
-	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
 	bisectString := getBisectStatusText(bisectStatus, bisectInfo)
 
+	hashString := commit.Hash
+	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
+	hashLength := common.UserConfig.Gui.CommitHashLength
+	if hashLength == 0 && !icons.IsIconEnabled() { // if no icons and no hash, show a char so user still sees status color
+		hashString = common.UserConfig.Gui.CommitStatusChar
+	} else if hashLength < len(hashString) {
+		hashString = hashString[:hashLength]
```

Comment:
I wanted the minimum of commitHashLength to be 0, but then you wouldn't be able to see the status color (if you didn't have icons on), so I added a configurable string that displays in that case.

Reply:
I agree with Jesse's first reaction, I think this is unnecessary. The main reason I dislike it is that it adds a new config option that only very few people will find useful. We should try to avoid adding new config options if we don't have to; we have too many already, and it makes it harder to find the really important ones.

One solution to this that I could live with is to hard-code the replacement character. I'd suggest `*` in that case; we already use it as a replacement character for the branch icon for stacked branches, and that one is not configurable either. I don't have a problem with using the same character for two different things, as they are colored differently:
<img width="327" alt="image" src="https://github.com/jesseduffield/lazygit/assets/1225667/6f9249fc-a83d-48de-b51e-05f45f812b23">

Another problem with your current implementation is that you get two spaces instead of one between the icon and the author if nerd fonts are used and the hash length is zero. The reason is that you are coloring the empty string, and our remove-empty-columns logic doesn't handle that. @jesseduffield I have run into this before, and I wonder if we should just call Decolorize when removing empty columns:
```diff
diff --git a/pkg/utils/formatting.go b/pkg/utils/formatting.go
index 139946ddb..e6279d350 100644
--- a/pkg/utils/formatting.go
+++ b/pkg/utils/formatting.go
@@ -81,7 +81,7 @@ func excludeBlankColumns(displayStringsArr [][]string, columnAlignments []Alignm
 outer:
 	for i := range displayStringsArr[0] {
 		for _, strings := range displayStringsArr {
-			if strings[i] != "" {
+			if Decolorise(strings[i]) != "" {
 				continue outer
 			}
 		}
```
We are already calling Decolorize in getPadWidths, so we may want to refactor the code to do it only once. However, I'm not sure if the heap allocations for allocating an array of decolorized strings are more or less expensive than doing the regex thing twice, so some profiling would be needed.

@oliviaBahr For now you could solve that problem by coloring the string only if it's non-empty.

Reply:
hard coding a value works for me.

Reply:
Sounds good, I removed the config option and '*' is hard coded 6ea35f4d5bf3aa5f62de359d9bd6bcd1c75f0a94